apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/modules/components/Mintable.tsx

```diff
@@ -0,0 +1,477 @@
+"use client";
+import { Spinner } from "@/components/ui/Spinner/Spinner";
+import {
+  Accordion,
+  AccordionContent,
+  AccordionItem,
+  AccordionTrigger,
+} from "@/components/ui/accordion";
+import { Alert, AlertTitle } from "@/components/ui/alert";
+import { Button } from "@/components/ui/button";
+import { Checkbox, CheckboxWithLabel } from "@/components/ui/checkbox";
+import {
+  Form,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import { Separator } from "@/components/ui/separator";
+import { Skeleton } from "@/components/ui/skeleton";
+import { Textarea } from "@/components/ui/textarea";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useMutation } from "@tanstack/react-query";
+import { CircleAlertIcon } from "lucide-react";
+import { useCallback } from "react";
+import { useForm } from "react-hook-form";
+import { toast } from "sonner";
+import { type PreparedTransaction, sendAndConfirmTransaction } from "thirdweb";
+import { MintableERC721, MintableERC1155 } from "thirdweb/modules";
+import { useReadContract } from "thirdweb/react";
+import type { NFTMetadataInputLimited } from "types/modified-types";
+import { parseAttributes } from "utils/parseAttributes";
+import { z } from "zod";
+import { addressSchema } from "../zod-schemas";
+import { ModuleCardUI, type ModuleCardUIProps } from "./module-card";
+import type { ModuleInstanceProps } from "./module-instance";
+import { AdvancedNFTMetadataFormGroup } from "./nft/AdvancedNFTMetadataFormGroup";
+import { NFTMediaFormGroup } from "./nft/NFTMediaFormGroup";
+import { PropertiesFormControl } from "./nft/PropertiesFormControl";
+
+export type UpdateFormValues = {
+  primarySaleRecipient: string;
+};
+
+// TODO - add form validation with zod schema for mint form
+
+export type MintFormValues = NFTMetadataInputLimited & {
+  useNextTokenId: boolean;
+  recipient: string;
+  amount: number;
+  supply: number;
+  customImage: string;
+  customAnimationUrl: string;
+  attributes: { trait_type: string; value: string }[];
+  tokenId?: string;
+};
+
+function MintableModule(props: ModuleInstanceProps) {
+  const { contract, ownerAccount } = props;
+
+  const primarySaleRecipientQuery = useReadContract(
+    MintableERC721.getSaleConfig,
+    {
+      contract: contract,
+    },
+  );
+
+  const isErc721 = props.contractInfo.name === "MintableERC721";
+  const isBatchMetadataInstalled = !!props.allModuleContractInfo.find(
+    (module) => module.name.includes("BatchMetadata"),
+  );
+  const isSequentialTokenIdInstalled = !!props.allModuleContractInfo.find(
```

Comment:
same thing here - `isSequentialTokenIdInstalled` is not used anywhere in Mintable module UI - why it is passed around?